server/models/server-service.js

```diff
@@ -70,6 +70,52 @@ module.exports = function extendServerService(ServerService) {
   }
   ServerService.setServiceCommit = setServiceCommit;
 
+  // XXX(sam) serviceFindOrCreate, serviceCreate, serviceFind were almost
+  // verbatim ripped from client/client.js: there should be a way of sharing the
+  // implementation, perhaps this code can be attached to ServerService in
+  // common/models and be available identically on client and server as a js
+  // method?
+  function serviceFindOrCreate(name, scale, callback) {
+    var self = this;
+    self.serviceFind(name, function(err, service) {
+      if (err && err.statusCode !== 404) return callback(err);
+      if (service) return callback(null, service);
+      self.serviceCreate(name, scale, callback);
+    });
+  }
+  ServerService.serviceFindOrCreate = serviceFindOrCreate;
+
+  function serviceCreate(name, scale, callback) {
+    var self = this;
```

Comment:
keeps lint happy, but I'll add an anti-lint comment and simplify the code. lint hates `Service = this`, it expects all this to be assigned to self...